torch/testing/_internal/distributed/fake_pg.py

```diff
@@ -44,6 +44,13 @@ def reduce_scatter(self, output_tensor, scatter_list, opts=ReduceScatterOptions(
         return ret_work(output_tensor)
 
     def _allgather_base(self, output_tensor, input_tensor, opts=AllgatherOptions()):
+        # assume each rank have the same input tensor so we just copy to the results
+        # since it's not a real allgather, we simply make this copying logic to let
+        # some simple validation works (i.e. calling allgather to see if each rank have
+        # the same tensor or not)
```

Comment:
can you make a more explicit comment here and also in the PR desc / commit msg, explaining that in general its not good form to try to make FakePG work with 'real data', but the reasoning here is that we want FakePG to work with DeviceMesh's initialization code which we feel is worth this tradeoff?  

(And suggest in general people should use MTPG for cases where they may care about real data from collectives)